src/main/kotlin/de/bund/digitalservice/useid/tenant/ResolveTenantFilter.kt

```diff
@@ -0,0 +1,35 @@
+package de.bund.digitalservice.useid.tenant
+
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.web.filter.OncePerRequestFilter
+
+internal const val REQUEST_ATTR_TENANT = "tenant"
+internal const val PARAM_NAME_TENANT_ID = "tenant_id"
+internal const val REQUEST_PATH_WIDGET = "/widget"
+
+class ResolveTenantFilter(
+    private val tenantProperties: TenantProperties,
+) : OncePerRequestFilter() {
+    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
+        val fallbackTenant = Tenant().apply {
+            id = "unknown"
+        }
+
+        val tenant =
+            if (request.servletPath.equals(REQUEST_PATH_WIDGET)) {
+                val host = request.getParameter("hostname") ?: ""
+                tenantProperties.findByAllowedHost(host) ?: fallbackTenant
+            } else if (request.getParameter(PARAM_NAME_TENANT_ID) != null) {
+                val id = request.getParameter(PARAM_NAME_TENANT_ID)
+                tenantProperties.findByTenantId(id) ?: fallbackTenant
+            } else {
+                fallbackTenant
+            }
+
+        // assign entire object, because the WidgetController needs access to multiple properties instead of only the id
```

Comment:
This logic might change in the future. I personally would remove this comment. The code would break on compile time if someone changes it, so no need for a warning imho.